modules/modals/addItem.ts

```diff
@@ -0,0 +1,11 @@
+import { addItemForm } from '../domElements';
+import { useModal } from './utils';
+
+const addItemModalManager = useModal({
+  modalId: 'modal',
+  openButtonId: 'openModalButton',
+  closeButtonId: 'closeModal',
+});
+
+// callback to reset the form whenever we close the modal
+addItemModalManager?.hideModal(() => addItemForm.reset());
```

Comment:
Not a requested change - just a shout out that this is a lot easier to work with when creating a new modal. Before we had large files getting bunches of DOM elements, and assigning functionality to them, and in each one this was all pretty similar (which makes sense, we want modals to work pretty much the same way in each instance). 

Of course we're still doing this, but with a nice reusable function that only requires the DOM element ids to be passed.